src/modm/platform/dma/samx7x/dma.cpp

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2023, Christopher Durand
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include <modm/platform/dma/dma.hpp>
+#include <modm/platform/clock/clockgen.hpp>
+
+namespace modm::platform
+{
+
+void Dma::initialize()
+{
+	ClockGen::enable<ClockPeripheral::Xdmac>();
+}
+
+auto Dma::channel(uint_fast8_t index) -> DmaChannel
```

Comment:
```suggestion
DmaChannel Dma::channel(uint_fast8_t index)
```
Consistency?

Reply:
same

Reply:
Nope, this one actually compiles :D

Reply:
You are right. `DmaChannel` is a class in `modm::platform`.